### ISSUE TITLE
Option to Deploy Across Multiple Regions / Data centres

### ISSUE THREAD

**Issue by taylora3** [Azure/Moodle#93]
I was on the call during the week. I would like a way to deploy across multiple regions. e.g have 2 Moodle deployments in an active/active set up e.g one in North Europe and one in West Europe.  The aim is to make sure that the set up is always on. So if the Data centre in North Europe goes down it will fall back to the one in West Europe and vice-versa. What is the projected uptime for the Maximal deployment? Many thanks and happy to work with you.

**Comment by taylora3** [comment]
@hosungsmsft You are right. I meant the SLA. e.g the 99.99% up-time referenced in [azure availability zones](https://docs.microsoft.com/en-us/azure/availability-zones/az-overview) We are looking at worst case scenarios and disaster recovery. If you could point me in the right direction of docs etc that would be appreciated.

**Comment by hosungsmsft** [comment]
Thanks for your clarification. Due to the "solution"-nature of our templates, we currently don't have a concrete SLA number of a deployed Moodle cluster as a whole. We may try to do some reliability analysis of an entire cluster from individual components' SLA numbers, but that may not be very accurate or useful. Component-wise SLAs are as follows for the NFS option (no file system high availability):

- Controller VM: 99.9%, as it's a single instance VM and according to [this](https://azure.microsoft.com/en-us/support/legal/sla/virtual-machines/v1_8/) Azure VM SLA documentation.
- VMSS VMs: 99.95%, as they are deployed in an Azure availability set. If they are deployed in availability zones (which are said to be in preview on select regions only), SLA will be 99.99% (see the same Azure VM SLA documentation above), but I don't think our templates currently support the option of deploying VMSS VMs in availability zones. We'll look into that.
- MySQL DB: 99.99% according to [this](https://azure.microsoft.com/en-us/support/legal/sla/mysql/v1_0/) Azure MySQL SLA documentation.

If customer chooses Gluster for file system high availability, an Azure availability set-based Gluster VM cluster is also deployed, whose SLA is 99.95%. The maximal deployment also deploys an Azure Redis cache, whose [published](https://azure.microsoft.com/en-us/support/legal/sla/cache/v1_0/) SLA is 99.9%. Please note that there are other components' SLAs that'll affect the SLA of the entire cluster as a whole. Hope this helps.

**Comment by hosungsmsft** [comment]
Thanks for sharing your global vnet idea. It looks to me, no matter what, this is going to be a non-trivial cloud solution architecting task that'll need significant research/experiences. Please do feel free to reach out to us by email if you'd like to have more in-depth discussion on that aspect with us (hosung.song@microsoft.com and ross.gardler@microsoft.com).
